src/lib/CanvasGroupings.js

```diff
@@ -26,27 +29,55 @@ export default class CanvasGroupings {
    * of canvases, while book view creates pairs.
    */
   groupings() {
-    if (this._groupings) { // eslint-disable-line no-underscore-dangle
-      return this._groupings; // eslint-disable-line no-underscore-dangle
+    if (this._groupings) {
+      return this._groupings;
     }
     if (this.viewType !== 'book') {
       return this.canvases.map(canvas => [canvas]);
     }
     const groupings = [];
+    const canvasGroupingMap = [];
+    let groupIndex = 0;
     this.canvases.forEach((canvas, i) => {
+      if (!groupings[groupIndex]) {
+        groupings[groupIndex] = [];
+      }
+
+      canvasGroupingMap[i] = groupIndex;
+
       if (i === 0) {
-        groupings.push([canvas]);
+        canvasGroupingMap[i] = groupIndex;
+        groupings[groupIndex].push(canvas);
+        groupIndex += 1;
         return;
       }
-      // Odd page
-      if (i % 2 !== 0) {
-        groupings.push([canvas]);
-      } else {
-        // Even page
-        groupings[Math.ceil(i / 2)].push(canvas);
+
+      const hint = canvas && (
+        (canvas.getBehavior && canvas.getBehavior())
+        || (canvas.getViewingHint && canvas.getViewingHint())
```

Comment:
If I understand the results of my debugger correctly, `canvas` is a  [Canvas Object from the `manifesto.js` package](https://github.com/IIIF-Commons/manifesto/blob/master/src/Canvas.ts). This doesn't have a `getViewingHint()` method (at least not in a ES Module build of version 4.2.0). But this finding answers the last question from your Checklist, `CanvasGroupings.js` get's called.